### PR TITLE
Issue/118 Refactor `decompose_flow_vectors` for further WPS processes

### LINKED ISSUE #118: Refactor `decompose_flow_vectors`
The [`decompose_flow_vectors`](https://github.com/pacificclimate/climate-explorer-data-prep/blob/master/scripts/decompose_flow_vectors) script has no function encapsulation with all of the code being at the top level and executed at import time.

Better practice would be to wrap the functionality in a `main()` function, so that it can be imported into other programs and run programmatically (e.g. in a WPS process).

Follow the example in [`generate_prsn`](https://github.com/pacificclimate/climate-explorer-data-prep/blob/master/scripts/generate_prsn) to do this.

### PR DESCRIPTION
**Pull request by sum1lim**
This PR resolves #118 

For a better utility of the program, main function has been created and wraps the functionality of `decompose_flow_vectors`. The script and main function will be imported into other programs, thus the `scripts` package should be included in `ce_dataprep`. In ordered to do that `__init__.py` has been created.

### CODE REVIEW

File: scripts/decompose_flow_vectors [hunk 1]
```diff
@@ -1,108 +1,61 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Process an indexed flow direction netCDF into a vectored netCDF suitable for ncWMS display"
+    )
+    parser.add_argument("source_file", metavar="infile", help="source netCDF file")
+    parser.add_argument("dest_file", metavar="outfile", help="destination netCDF file")
+    parser.add_argument(
+        "variable", metavar="variable", help="netCDF variable describing flow direction"
+    )
+
```
Comment (nikola-rados) [inline scripts/decompose_flow_vectors:59]: We probably want to add a flag for log level.

File: setup.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
         scripts/generate_prsn
         scripts/split_merged_climos
         scripts/update_metadata
+	scripts/decompose_flow_vectors
```
Comment (nikola-rados) [inline setup.py:34]: Just need two more spaces for this item to line up with the others in the list.

File: dp/decompose_flow_vectors.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+#!python
```
Comment (nikola-rados) [inline dp/decompose_flow_vectors.py:1]: This entire file looks great!

**Comment by corviday** [comment]
This script was [excluded from CI testing](https://github.com/pacificclimate/climate-explorer-data-prep/issues/96) because the way it was written made it difficult for the test system to call. That issue should be fixable now, but I'm not sure if we'd rather make it part of this PR, or do it separately.

**Comment by rod-glover** [comment]
Thanks, @corviday for providing the ground-truth check.

**Comment by sum1lim** [comment]
@corviday It is really interesting to see the actual output illustrated on a real map and glad to see the outputs are correct. Thank you! I guess this PR is ready to be merged, so I will do it right away. Thanks to all who made comments!

**Pull request merged**